pkg/machine/provider/platform_darwin.go

```diff
@@ -36,3 +41,45 @@ func Get() (vmconfigs.VMProvider, error) {
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
 }
+
+// SupportedProviders returns the providers that are supported on the host operating system
+func SupportedProviders() []define.VMType {
+	return []define.VMType{define.AppleHvVirt}
+}
+
+// InstalledProviders returns the supported providers that are installed on the host
+func InstalledProviders() ([]define.VMType, error) {
+	var outBuf bytes.Buffer
+	// Apple's Virtualization.Framework is only supported on MacOS 11.0+
+	const SupportedMacOSVersion = 11
+
+	cmd := exec.Command("sw_vers", "--productVersion")
+	cmd.Stdout = &outBuf
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("unable to check current macOS version using `sw_vers --productVersion`: %s", err)
+	}
+
+	// the output will be in the format of MAJOR.MINOR.PATCH
```

Comment:
This code is similar to what crc is doing https://github.com/crc-org/crc/blob/main/pkg/os/darwin/release_info.go
Maybe using a `semver` package would make the version parsing/checking simpler.